tests/gis_tests/geos_tests/test_geos.py

```diff
@@ -241,6 +242,93 @@ def test_eq_with_srid(self):
+        """
+        # empty inputs of different types are not equals_identical
+        g1 = fromstr("POINT EMPTY")
+        g2 = fromstr("LINESTRING EMPTY")
+        self.assertFalse(g1.equals_identical(g2))
+
+        # empty inputs of different dimensions are not equals_identical
+        g1 = fromstr("POINT EMPTY")
+        g2 = fromstr("POINT Z EMPTY")
+        self.assertFalse(g1.equals_identical(g2))
+
+        # non empty inputs of different dimensions are not equals_identical
+        g1 = fromstr("POINT Z (1 2 3)")
+        g2 = fromstr("POINT M (1 2 3)")
+        self.assertFalse(g1.equals_identical(g2))
+
+        g1 = fromstr("POINT ZM (1 2 3 4)")
+        g2 = fromstr("POINT Z (1 2 3)")
+        self.assertFalse(g1.equals_identical(g2))
+
+        # inputs with different structure are not equals_identical
+        g1 = fromstr("LINESTRING (1 1, 2 2)")
+        g2 = fromstr("MULTILINESTRING ((1 1, 2 2))")
+        self.assertFalse(g1.equals_identical(g2))
+
+        # inputs with different types are not equals_identical
+        g1 = fromstr("GEOMETRYCOLLECTION (LINESTRING (1 1, 2 2))")
+        g2 = fromstr("MULTILINESTRING ((1 1, 2 2))")
+        self.assertFalse(g1.equals_identical(g2))
+
+        # input with identical non-finite values
+        g1 = Point(x=float("nan"), y=math.inf)
+        g2 = Point(x=float("nan"), y=math.inf)
+        self.assertTrue(g1.equals_identical(g2))
+
+        # equal lines
+        g1 = fromstr("LINESTRING M (1 1 0, 2 2 1)")
+        g2 = fromstr("LINESTRING M (1 1 0, 2 2 1)")
+        self.assertTrue(g1.equals_identical(g2))
+
+        # different lines
+        g1 = fromstr("LINESTRING M (1 1 0, 2 2 1)")
+        g2 = fromstr("LINESTRING M (1 1 1, 2 2 1)")
+        self.assertFalse(g1.equals_identical(g2))
+
+        # equal polygons
+        g1 = fromstr("POLYGON ((0 0, 1 0, 1 1, 0 0))")
+        g2 = fromstr("POLYGON ((0 0, 1 0, 1 1, 0 0))")
+        self.assertTrue(g1.equals_identical(g2))
+
+        # different polygons
+        g1 = fromstr("POLYGON ((0 0, 1 0, 1 1, 0 0))")
+        g2 = fromstr("POLYGON ((1 0, 1 1, 0 0, 1 0))")
+        self.assertFalse(g1.equals_identical(g2))
+
+        # different polygons (number of holes)
+        g1 = fromstr("POLYGON ((0 0, 10 0, 10 10, 0 10, 0 0), (1 1, 2 1, 2 2, 1 1))")
+        g2 = fromstr(
+            "POLYGON ((0 0, 10 0, 10 10, 0 10, 0 0), (1 1, 2 1, 2 2, 1 1), "
+            "(3 3, 4 3, 4 4, 3 3))"
+        )
+        self.assertFalse(g1.equals_identical(g2))
+
+        # identical collections
+        g1 = fromstr("MULTILINESTRING ((1 1, 2 2), (2 2, 3 3))")
+        g2 = fromstr("MULTILINESTRING ((1 1, 2 2), (2 2, 3 3))")
+        self.assertTrue(g1.equals_identical(g2))
+
+        #  different collections (structure)
+        g1 = fromstr("MULTILINESTRING ((1 1, 2 2), (2 2, 3 3))")
+        g2 = fromstr("MULTILINESTRING ((2 2, 3 3), (1 1, 2 2))")
+        self.assertFalse(g1.equals_identical(g2))
```

Comment:
Can we move `Point(x=float("nan")` to a separate test method, and use `subTest()` for other cases?, e.g.
```python
def test_equals_identical(self):
    tests = [
        ("POINT EMPTY", "LINESTRING EMPTY"),
        ("POINT EMPTY", "POINT Z EMPTY"),
        ...
    ]
    for g1, g2 in tests:
        with self.subTest(g1=g1, g2=g2):
            self.assertIs(g1.equals_identical(g2), False)
```